Convertopackage

Valirou opened the pull request:
I converted most of the main_word_analysis code to an wordanalysis package. Before going further, I wanted to have your review on it. I want to be sure that I'm on the right way. 
There is a part in the main code with 5 comments separated by newline those are ideas for new functions.
I did not pass the code into PEP8 yet.

Valirou committed:
Prepared wordanalysis package

According to (I stopped at "Generating distribution archives") : https://packaging.python.org/tutorials/packaging-projects/

:alarm_clock:  TO DO : CHOOSE A LICENSE :alarm_clock:

Valirou committed:
Added imports needed in the new wordanalysis package + modified preprocess

I also modified preprocess function to include the creation of the stop words list because it was needed in it.

Valirou committed:
Modified the main code to use the wordanalysis package

I created a __init__ file to tell that it is a package but I still don't know what to do with it

Valirou committed:
Added import in wordanalysis __init__

Valirou committed:
Created Rawtxt class with preprocess as method in FunctionTest

I modified the preprocess to add some options. By doing so, the
function can be used with differents intentions as per getting a
clean text in capitals or in lowercase.

Valirou committed:
Created the class WOSdf in FunctionTest

This class permit to go even faster in accessing columns of WOS exported database

Valirou committed:
Created the class Process in FunctionTest

I rassemble all the function that produce xxx to analyse the text. I also converted the section that sorted the article by score into a function then I added it in the Process class but I remove the part that sorted it because it make it easier to do this proscess twice...

Valirou committed:
Added functions in Process() and created RawDf class in FunctionTest

I added :
	most_common_bigrams
	most_common_trigrams
	nb_pub_by_year (modified it because it was causing bar overlapping when used second in a script with multiple plot)
	score
I created RawDf to remove duplicates, empty columns and change NaN for string (string needed for my words analysis)

Valirou committed:
Oups see last commit
(missing changes due to beginner(me) using prompt command)

Valirou committed:
Added tested classes and __init__ file in wordanalysis package

I inserted in the package all the classes that I have tested. I have also modified the main code to use this new package.
Now I see that I have a lot of pycache, next step is to make a gitignore for them.

Valirou committed:
Removed pycache and added them to .gitignore

gitignore took from dataming branch

ilyasst commented:
Pour pouvoir installer le package, je pense qu'il manque un fichier `setup.py` tel que décrit dans ce commentaire (https://stackoverflow.com/a/61762525). Il est ensuite possible d'installer le package avec `python setup.py develop`.
Après avoir ajouté ce fichier, je ne parviens tout de même pas à installer le package. Voir l'exemple suivant pour la structure de fichiers: [](https://measurespace.medium.com/how-to-build-your-first-python-package-4e28ea32208b)
Je peux revoir après ça.

Valirou committed:
Restructuring of wordanalysis package according to PyPA

Chose MIT license : add it in LICENSE file and remove comments in setup.cfg
Updated his README but the links are improper
Added a setup.py even though it has a .cfg because we do not want to publish it at PyPi so it needs a .py to run it locally

ilyasst committed:
Merge pull request #2 from jaklengaigne/convertopackage

Restructuring of wordanalysis package according to PyPA

ilyasst committed:
Necessary file structure changes; and adjustments to setup.cfg

ilyasst committed:
Adjustments to main_word_analysis so it can be used to test the package

ilyasst committed:
Added a requirements.txt to be able to install all the necessary dependencies with pip or conda easily

ilyasst commented:
Was handled in PR #2
